compiler/src/dotty/tools/dotc/transform/Pickler.scala

```diff
@@ -55,35 +57,37 @@ class Pickler extends Phase {
     }
     {
       val pickler = new TastyPickler(cls)
-      if (ctx.settings.YtestPickler.value) {
+      if ctx.settings.YtestPickler.value then
         beforePickling(cls) = tree.show
         picklers(cls) = pickler
-      }
       val treePkl = pickler.treePkl
       treePkl.pickle(tree :: Nil)
-      treePkl.compactify()
-      pickler.addrOfTree = treePkl.buf.addrOfTree
-      pickler.addrOfSym = treePkl.addrOfSym
-      if (tree.span.exists)
-        new PositionPickler(pickler, treePkl.buf.addrOfTree).picklePositions(tree :: Nil)
-
-      if (!ctx.settings.YdropComments.value)
-        new CommentPickler(pickler, treePkl.buf.addrOfTree).pickleComment(tree)
-
-      // other pickle sections go here.
-      val pickled = pickler.assembleParts()
-      unit.pickled += (cls -> pickled)
-
-      def rawBytes = // not needed right now, but useful to print raw format.
-        pickled.iterator.grouped(10).toList.zipWithIndex.map {
-          case (row, i) => s"${i}0: ${row.mkString(" ")}"
-        }
-
-      // println(i"rawBytes = \n$rawBytes%\n%") // DEBUG
-      if (pickling ne noPrinter) {
-        println(i"**** pickled info of $cls")
-        println(new TastyPrinter(pickled).printContents())
-      }
+      val pickledF = Future {
```

Comment:
Is this really going to parallelize things? Or should we also surround some parts with `concurrent.blocking`?

Reply:
Why would it not parallelize? I thought a `Future {...}` spans a separate computation. From the 2.13 Future API:
```
final def apply[T](body: => T)(implicit executor: ExecutionContext): Future[T]

Starts an asynchronous computation and returns a Future instance with the result of that computation.
...
The result becomes available once the asynchronous computation is completed.
```
I don't know `concurrent.blocking`. Can you explain what it would do?

Reply:
When you have several `Future { ... }` computations, the only thing you can tell is that they will run concurrently. Whether or not they will be run in parallel depends on the behavior of the underlying `ExecutionContext`. In this regard, the behavior of `ExecutionContext.global` does not parallelize `Future { ... }` blocks unless they are marked as `blocking`. See https://github.com/scala/bug/issues/12089 for a (longer) discussion.

By the way I just see that in your case you `Await` for the `Future` to be completed *within* the `for` loop, which means that these computations can’t be executed in parallel anyway. You probably want to create a list of `Future` and then await after evaluating that list has been completed?

Reply:
>  See scala/bug#12089 for a (longer) discussion.

Isn't this issue about nested Future blocks? Here we don't do any nesting so I'd expect the default executioncontext to run up to n Future in parallel where n is the number of processor cores.

> By the way I just see that in your case you Await for the Future to be completed within the for loop,

Which for loop are you refering to? Await is called from `force()`, a reference to `force` is stored in `pickled` and it is then executed much later in the backend

Reply:
> Isn't this issue about nested Future blocks?

True.

> Here we don't do any nesting so I'd expect the default executioncontext to run up to n Future in parallel where n is the number of processor cores.

That’s what I’d expect too, but we should check :)

> Await is called from `force()`, a reference to `force` is stored in `pickled` and it is then executed much later in the backend

Oh, thanks for clarifying, I thought `force` was a *call* (without parenthesis), not an eta-expansion. So, forget what I told about the loop :)

Reply:
I just verified by adding some debug output. It runs indeed in parallel.